<launch>
  <!-- ============ Arguments ============ -->
  <!-- FrameData -->
  <arg name="bag_file" default="$(env HOME)/datasets/uHumans/khronos_office_2.bag"/>
  <arg name="semantic_color_mapping" default="$(find hydra_ros)/config/color/uhumans2_apartment.csv"/>
  <arg name="play_rate" default="1"/>
  <arg name="sim_time_required" default="true"/>
  <arg name="use_gt_frame" default="true"/>
  <arg name="use_prerecorded_objects" default="false"/>
  <arg name="use_oneformer" default="false"/>
  <arg name="semantic_image_topic" default="/tesse/seg_cam/rgb/image_raw" unless="$(arg use_oneformer)"/>
  <arg name="semantic_image_topic" default="semantic_color/image_raw" if="$(arg use_oneformer)"/>
  <arg name="paused" default="false"/>

  <!-- ============ Run ============ -->
  <!-- Auto generated args -->
  <arg name="clock_arg" value="" unless="$(arg sim_time_required)"/>
  <arg name="clock_arg" value="--clock" if="$(arg sim_time_required)"/>
  <param name="use_sim_time" value="$(arg sim_time_required)" />
  <arg name="paused_arg" value="" unless="$(arg paused)"/>
  <arg name="paused_arg" value="--pause" if="$(arg paused)"/>

  <!-- Play the bag. -->
  <node name="dataset" pkg="rosbag" type="play" args="$(arg bag_file) -r $(arg play_rate) $(arg clock_arg) $(arg paused_arg)">
    <remap unless="$(arg use_gt_frame)" from="/tf" to="/tf_null"/>
  </node>

  <include if="$(arg use_oneformer)" file="$(find semantic_recolor)/launch/semantic_recolor.launch" pass_all_args="true">
    <arg name="prerecorded_topic" value="/tesse/oneformer/labels/image_raw"/>
    <arg name="labelspace_name" value="ade20k_mp3d"/>
    <arg name="use_prerecorded_semantics" value="true"/>
  </include>

  <!-- Static TF transforms. -->
  <include file="$(find hydra_ros)/launch/static_tfs/uhumans2_static_tfs.xml"/>

</launch>
